minio/policy.py

```diff
@@ -477,23 +477,24 @@ def get_policy(statements, bucket_name, prefix=''):
     obj_read_only = False
     obj_write_only = False
     for s in statements:
-        resource = s['Resource']
-        if (resource == object_resource or
-                fnmatch.fnmatch(object_resource, resource)):
-            read_only, write_only = _get_object_policy(s)
-            if len(matched_resource) < len(resource):
-                obj_read_only = read_only
-                obj_write_only = write_only
-                matched_resource = resource
-            elif len(matched_resource) == len(resource):
-                obj_read_only = obj_read_only or read_only
-                obj_write_only = obj_write_only or write_only
-                matched_resource = resource
-        elif resource == bucket_resource:
-            common_found, read_only, write_only = _get_bucket_policy(s, prefix)
-            bucket_common_found = bucket_common_found or common_found
-            bucket_read_only = bucket_read_only or read_only
-            bucket_write_only = bucket_write_only or write_only
+        resources = _get_resource(s)
+        for resource in resources:
+            if (resource == object_resource or
+                    fnmatch.fnmatch(object_resource, resource)):
```

Comment:
> Have a function to work with single resource and use that function in a for loop.

That function needs a lot of context that needs to be passed. Won't that make ugly?

@harshavardhana Yes, its still WIP.